src/pages/extending-chat-widget/chat-widget-adapters/index.mdx

```diff
@@ -0,0 +1,763 @@
+| **[Vue](#vue)**     | [@livechat/widget-vue](https://www.npmjs.com/package/@livechat/widget-vue)         |
+| **[Angular](#vue)** | [@livechat/widget-angular](https://www.npmjs.com/package/@livechat/widget-angular) |
+
+# Universal Adapter model
+
+Each adapter library implements the concept of integrating with Chat Widget JS API is a very similar way.
+Various implemnetation detials of each framework makes them different of course but they share common lifecycle model and reactive data interface concept.
+The core implementation shared between each framework's specific adapter is distrubuted as a separate library [`@livechat/widget-core`](https://www.npmjs.com/package/@livechat/widget-core).
+
+## Component Lifecycle
+
+### Mount
+
+During the mounting phase of the component, the Chat Widget is fully initialized. As before the mount time widget instance does not exist as also does not have access to any provided data, the mount time is the first moment when an instance of the widget is created and it receives the first set of data.
+
+1. Use obtained data as parameters for widget initialization
+
+2. Create Chat Widget JS API instance similarly to a snippet code
+
+3. Register provided event handlers for the corresponding events
+
+4. Notify reactive data interface that widget has been initialized to update related data
+
+### Update
+
+This phase allows updating provided data or callbacks by changing their value or reference. In order to do so, the component internally observes for such changes and translate them to appropriate imperative mutations via JS API method calls.
+
+1. If properties were updated then call coresponding setter methods on JS API
+
+2. If properties that require full reload were updated then reinitialize widget with new properties
+
+3. If event handlers were updated then unregister the previous one and register a new one for the corresponding event
+
+### Unmount
+
+This phase results in complete Chat Widget cleanup but allowing for proper Mount phase once again.
+
+1. Unregister all registered callbacks
+
+2. Call destroy method from JS API
+
+3. Notify reactive data interface that widget has been destroyed to clear related data
+
+## Reactive data interface
+
+After looking into existing JS API events and available getters we can distinguish a set of entities that the Chat Widget exposes to interact with it.
+Each entity contains data that may change during Chat Widget lifetime.
+Entites are properly initialized on widget load and cleared after widget is destroyed.
+All posible updates are being notified for the consumer of those data.
+The consumer have a way to subscirbe or unsubscribe for the updates in any given moment during application lifetime.
+The Reactive data interface efectivly impements the Observer pattern in which a subject maintains a list of its observers and notifies them about any state changes, usually by calling one of their methods.
```

Comment:
```suggestion
The Reactive data interface effectively implements the Observer pattern in which a subject maintains a list of its observers and notifies them about any state changes, usually by calling one of their methods.
```

Maybe we could link a _good_ description of the observer pattern here if a reader would like to learn more quickly?

Reply:
the reactive data interface has already been mentioned and wasn't capitalized

Reply:
@klarzynskik good point 👍  will look for something good 👀

Reply:
@klarzynskik added 😉